File: 1071/go/Solution.go
package Solution1071

func solution(str1 string, str2 string) string {
	var min_len int
	if len(str1) < len(str2) {
		min_len = len(str1)
	} else {
		min_len = len(str2)
	}
	for i := min_len; i > 0; i-- {
		if len(str1)%i == 0 && len(str2)%i == 0 {
			if str1[:i] == str2[:i] && multiplyString(str1[:i], len(str1)/i) == str1 && multiplyString(str2[:i], len(str2)/i) == str2 {
				return str1[:i]
			}
		}
	}
	return ""
}

func multiplyString(str string, n int) string {
	result := ""
	for i := 0; i < n; i++ {
		result += str
	}
	return result
}

File: 1071/python/solution.py
def solution(str1: str, str2: str) -> str:
    len_str1 = len(str1)
    len_str2 = len(str2)
    min_len: int = min(len_str1, len_str2)
    for i in range(min_len, 0, -1):
        sub_str1 = str1[:i]
        sub_str2 = str2[:i]
        if sub_str1 == sub_str2:
            if len_str1 % i == 0 and len_str2 % i == 0:
                if str1 == sub_str1 * (len_str1 // i) and str2 == sub_str2 * (len_str2 // i):
                    return sub_str1
    return ""
